### PR TITLE
Define what we mean by 'scalability'.

### PR DESCRIPTION
**Pull request by bangerth**
I have found that when rewriting step-17, I kept avoided such a definition. I think
it is important to understand this concept, so here is a lengthy description.

### CODE REVIEW

File: doc/doxygen/headers/glossary.h [hunk 1]
```diff
@@ -1300,6 +1300,150 @@
...
+ * every processor) inversely proportional to $P$ by just throwing more
+ * processors at the problem. In particular, strong scalability implies
+ * that if we provide twice as many processors, then run time and memory
+ * consumption on every process will be reduced by a factor of two. In
+ * other words, we can solve the <i>same problem</i> faster and faster
+ * by providing more and more processors.
+ *
+ * Conversely, "weak scalability" means that if we increase the problem
+ * size $N$ by a fixed factor, and increase the number of processors
+ * $P$ available to solve the problem by the same factor, that the
```
Comment (kronbichler) [inline doc/doxygen/headers/glossary.h]: that -> then (?) (I'm not a native speaker but that sounds really strange to me at this place, given that you use it in the dependent clause already.)

File: doc/doxygen/headers/glossary.h [hunk 2]
```diff
@@ -1300,6 +1300,150 @@
...
+ *   there are parts of the program whose CPU time requirements do
+ *   not decrease with the number of processors $P$ you allocate for
+ *   a fixed size $N$. In reality, the situation is actually even
+ *   worse: the more processes are participating in a communication
+ *   step, the longer it will generally take, for example because
+ *   the one process that has to add everyone's contributions has
+ *   to add everything up, requiring ${\cal O}(P)$ time. In other words,
+ *   CPU time <i>increases</i> with the number of processes, therefore
+ *   not only preventing a program from scaling strongly, but also from
+ *   scaling weakly.
```
Comment (kronbichler) [inline doc/doxygen/headers/glossary.h]: I would be a bit more cautious with this text. MPI implementations make great effort to efficiently implement Allreduce operations. Good implementations most likely do pairwise summation in a way that suits the network topology. You can expect to get O(log(P)) effort even on the most heavily 'loaded' processors. However, the problem really is communication latency of O(log(P)) where each communication step is around 1us for one hop (communication between two sites in the network). Thus, I would not explicitly state that there is one processor adding up *all* results but rather that even in the best case the sum will finally be computed on one processor (e.g. adding up two half sums) only and then needs to be communicated to everyone else again, so you have to go through the network O(log(P)) times forward and backward at the very least.

File: doc/doxygen/headers/glossary.h [hunk 3]
```diff
@@ -1300,6 +1300,150 @@
...
+ *   T = {\cal O}\left(\alpha \frac WP + (1-\alpha)W \right).
+ * @f}
+ * Consequently, the "speedup" you get, i.e., the factor by which your
+ * programs run faster on $P$ processors compared to running the program
+ * on a single process (assuming this is possible) would be
+ * @f{align*}
+ *   S = \frac{W}{\alpha \frac WP + (1-\alpha)W}
+ *     = \frac{P}{\alpha + (1-\alpha)P}.
+ * @f}
+ * If $\alpha>1$, which it is for all practically existing programs,
```
Comment (kronbichler) [inline doc/doxygen/headers/glossary.h]: Write `\alpha<1`.

File: doc/doxygen/headers/glossary.h (hunk 2 above)
Comment (bangerth) [inline doc/doxygen/headers/glossary.h]: Yes, fair point. I didn't want to make it more complicated than necessary -- the point is not that run time grows like O(P), but that it grows at all. I've added a parenthetical remark to this now.

**Comment by bangerth** [comment]
Thanks, @kronbichler . I've made the 3 changes you suggested.

**Comment by kronbichler** [comment]
Very nice contribution. I suggest to wait if someone else has remarks. (Please merge yourself if nobody speaks up within a day.)

**Comment by luca-heltai** [comment]
Very nice.

**Pull request merged**